src/intro_movie_scene_1.c

```diff
@@ -0,0 +1,431 @@
+#include "gf_gfx_loader.h"
+#include "global.h"
+#include "intro_movie_internal.h"
+#include "unk_0200ACF0.h"
+#include "unk_0200B150.h"
+#include "unk_0200FA24.h"
+#include "unk_020215A0.h"
+#include "unk_02022588.h"
+#include "system.h"
+#include "demo/opening/gs_opening.naix"
+
+// Copyright, Gamefreak logo, and sunrise
+
+void IntroMovie_Scene1_VBlankCB(void *pVoid);
+void IntroMovie_Scene1_Init(IntroMovieOvyData *data, IntroMovieScene1Data *sceneData);
+BOOL IntroMovie_Scene1_Main(IntroMovieOvyData *data, IntroMovieScene1Data *sceneData, int a2);
+void IntroMovie_Scene1_Exit(IntroMovieOvyData *data, IntroMovieScene1Data *sceneData);
+void IntroMovie_Scene1_InitBgs(IntroMovieOvyData *data);
+void IntroMovie_Scene1_LoadBgGfx(BgConfig *bgConfig);
+void IntroMovie_Scene1_SetBrightnessAndBgMaskColor(IntroMovieOvyData *data);
+void IntroMovie_Scene1_LoadSpriteGfx(IntroMovieOvyData *data, IntroMovieScene1Data *sceneData);
+void IntroMovie_Scene1_DestroySpritesAndObjectGfx(IntroMovieOvyData *data, IntroMovieScene1Data *sceneData);
+void IntroMovie_Scene1_CreateSprites(IntroMovieOvyData *data, IntroMovieScene1Data *sceneData);
+
+const u8 sIntroMovieScene1SpriteResCounts[4] = {1, 1, 1, 1};
+
+BOOL IntroMovie_Scene1(IntroMovieOvyData *data, void *pVoid) {
+    IntroMovieScene1Data *sceneData = (IntroMovieScene1Data *)pVoid;
+
+    if (IntroMovie_GetIntroSkippedFlag(data)) {
+        sceneData->state = 2;
+    }
+
+    switch (sceneData->state) {
+    case 0:
+        IntroMovie_Scene1_Init(data, sceneData);
+        ++sceneData->state;
+        break;
+    case 1:
+        if (IntroMovie_Scene1_Main(data, sceneData, IntroMovie_GetTotalFrameCount(data))) {
+            ++sceneData->state;
+        }
+        break;
+    case 2:
+        IntroMovie_Scene1_Exit(data, sceneData);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+void IntroMovie_Scene1_VBlankCB(void *pVoid) {
+    IntroMovieOvyData *data = (IntroMovieOvyData *)pVoid;
```

Comment:
rename pVoid to pData or something more descriptive

Reply:
but what if i want to point into the void?

Reply:
then do it on your own time

Reply:
won't-fix